emukit/core/loop/candidate_point_calculators.py

```diff
@@ -37,3 +38,51 @@ def compute_next_points(self, loop_state: LoopState) -> np.ndarray:
         """
         x, _ = self.acquisition_optimizer.optimize(self.acquisition)
         return np.atleast_2d(x)
+
+
+class MultiSourceSequential(CandidatePointCalculator):
+    """
+    This candidate point calculator optimizes the acquisition function separately for each information source,
+    then chooses the source with the highest acquisition value
+    """
+    def __init__(self, acquisition: Acquisition, acquisition_optimizer: AcquisitionOptimizer, space: ParameterSpace) -> None:
+        """
+        :param acquisition: Acquisition function to find maximum of
+        :param acquisition_optimizer: Optimizer of the acquisition function
+        :param space: Domain to search for maximum over
+        """
+        self.acquisition = acquisition
+        self.acquisition_optimizer = acquisition_optimizer
+        self.space = space
+        self.source_parameter = self._get_source_parameter()
+        self.n_sources = np.array(self.source_parameter.domain).size
+
+    def _get_source_parameter(self) -> InformationSourceParameter:
+        """
+        :return: The parameter containing the index of the information source
+        """
+        source_parameter = [param for param in self.space.parameters if isinstance(param, InformationSourceParameter)]
+        if len(source_parameter) == 0:
```

Comment:
Maybe we should move this logic to parameter space initialization? I find it hard to imagine a valid use case of multiple information source parameters.

Reply:
So we would have a `get_information_source_parameter()` method on `ParameterSpace`? I can do that

Reply:
Sorry, I meant validation logic, not that method. I don't think we need a method like this